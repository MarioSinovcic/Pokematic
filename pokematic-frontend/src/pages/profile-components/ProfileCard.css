.profile-card-shape {
    display: flex;
    flex-direction: row;
    height: 640px;
    padding: 25px 25px 0 25px;
    margin-top: 50px;
    margin-right: 100px;
    justify-content: center;
}

.profile-headers {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-content: center;
    height: 100%;
}

.profile-edge {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.profile-base-shape {
    width: 400px;
    height: 640px;
    background: #292929;

}

.profile-bottom {
    width: 100px;
    height: 640px;
    background:#292929;

}

.profile-cut {
    height: 0px;
    border-right: 100px solid transparent;
    border-bottom: 100px solid #292929;
}

.profile-card-shadow {
    background-color: #484747;
    height: 640px;
    width: 25px;
    z-index: 5;
}

.user-name {
    font-family: 'pkmn_rbygscregular' !important;
    color: white;
    font-size: 30px !important;
    text-align: center;
    width: 120%;
}

.user-image{
    width: 100%;
    border-radius: 300px;
}

.user-icon{
    /* position: absolute; */
    align-self: center;
    justify-self: center;
    width: 275px;
    height: 275px;  
    margin-bottom: 30px;
    margin-top: 40px;
 
    background-color: whitesmoke;
    background-size: 280px 280px;
    background-position: center;
    border-radius: 300px;
    border: white 5px solid;
 
    transition: 0.5s;
 }

.info {
    margin-bottom: 50px;
    margin-left: 80px;
    display: flex;
    flex-direction: column;
     align-items: center;
}


.role-button {
    size: 200px;
    background-color: #F64747 !important;
    color: white !important;
    font-family: 'pkmn_rbygscregular' !important;
    font-size: 20px !important;
    border-radius: 10px !important;
    height: 80px !important;

    align-self: center;
    padding: 10px !important;
    margin-bottom: 50px !important;
    margin-left: 80px !important;
}

