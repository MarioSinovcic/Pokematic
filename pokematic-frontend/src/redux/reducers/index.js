// import { combineReducers } from 'redux';

// calling the default reducer to create a link
import pokemonReducer from './pokemonReducer';
import { persistReducer } from 'redux-persist';

// Local storage in window browser for persistence
import storage from 'redux-persist/lib/storage';


const persistConfig = {
    // Where in the reducer do we want to start storing?
    key: 'root',
    storage,
  }


/**
 * Add this code once you have multiple reducers
 */

// const rootReducers = combineReducers({
//     // add reducer files references here
//     defaultReducer
// });


const rootReducers = pokemonReducer;

export default persistReducer(persistConfig, rootReducers);