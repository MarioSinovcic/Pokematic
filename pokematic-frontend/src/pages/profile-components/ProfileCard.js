import React from 'react';
import './ProfileCard.css';
import { Typography, Button } from '@material-ui/core';

function ProfileCard(props) {

    return (
        <div className="profile-card-shape">
        <div className="profile-card-shadow"></div>
            <div className="profile-base-shape">
            <div className="profile-headers">
                <div className="info">
                    <div className="user-icon">
                        <img className="user-image" src={props.picture} alt={"User Profile"}/>
                    </div>
                    <Typography className="user-name">{props.nickname}</Typography>
                </div>
                <Button className="sign-out-button" onClick={props.signOut}>Sign out</Button>
            </div>
    
            </div>
            <div className="profile-edge">
                <div className="profile-cut"></div>
                <div className="profile-bottom"></div> 
            </div>
        </div>
    )
}

export default ProfileCard;