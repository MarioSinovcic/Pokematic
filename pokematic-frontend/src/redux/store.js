import { createStore } from 'redux';
import rootReducers from './reducers';
import { persistStore } from 'redux-persist';


// creating and persist store for the redux structure
const store = createStore(
    rootReducers,
);
const persistor = persistStore(store);

export  { store, persistor};