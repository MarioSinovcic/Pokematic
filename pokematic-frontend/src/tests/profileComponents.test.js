import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import rootReducers from '../reducers';
import ProfileCard from '../pages/profile-components/ProfileCard';



describe('profile components', () => {

    let store;
    beforeEach(() => {
        store = createStore(rootReducers)
    })


    describe('snapshot matching', () => {

        it('Profile Card matches snapshot' , () => {
            const profileCard = renderer.create(<Provider store={store}>
                <ProfileCard />
            </Provider>).toJSON();
            expect(profileCard).toMatchSnapshot();
        });

    })


    




});