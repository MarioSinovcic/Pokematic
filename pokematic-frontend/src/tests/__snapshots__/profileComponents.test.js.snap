// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile components snapshot matching Profile Card matches snapshot 1`] = `
<div
  className="profile-card-shape"
>
  <div
    className="profile-card-shadow"
  />
  <div
    className="profile-base-shape"
  >
    <div
      className="profile-headers"
    >
      <div
        className="info"
      >
        <div
          className="user-icon"
        >
          <img
            alt="User Profile"
            className="user-image"
          />
        </div>
        <p
          className="MuiTypography-root user-name MuiTypography-body1"
        />
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text sign-out-button"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Sign out
        </span>
      </button>
    </div>
  </div>
  <div
    className="profile-edge"
  >
    <div
      className="profile-cut"
    />
    <div
      className="profile-bottom"
    />
  </div>
</div>
`;
